### ISSUE TITLE
bigints allow type overflow when encoding

### ISSUE THREAD

**Issue by 8de2fdb0** [stephenh/ts-proto#925]
I have an issue with following generated snippet:

```ts
 encode(message: MyMessage, writer: _m0.Writer = _m0.Writer.create()): _m0.Writer {
    if (message.ref !== BigInt("0")) {
      writer.uint32(9).fixed64(message.ref.toString());
    }
  },
```
It uses `--ts_proto_opt=forceLong=bigint` and uses bigints as input.

If I use following number as input for the `fixed64` the encoded protobuf value is 0
```
18_446_744_073_709_551_615n + 1n
```

So it basically overflows. 

Is there any way to guard against this unexpected behaviour?

**Comment by jcready** [comment]
Are you expecting an error to be thrown or for it to just not encode the zero value (after overflow) because it's the default value?.

**Comment by 8de2fdb0** [comment]
Good question, I could not find compiler spec for protoc and the protoc c++ implementation dos exactly the same, so I would consider that the correct behaviour.

Are there any guards for overflows or malformed input used in ts-proto currently? 

Maybe we should keep it this way and the user has to validate input before encoding.

**Comment by stephenh** [comment]
We have somewhat similar detection for Number/long overflows:

https://github.com/stephenh/ts-proto/blob/main/src/main.ts#L505

```
function longToNumber(long: Long): number {
  if (long.gt(globalThis.Number.MAX_SAFE_INTEGER)) {
    throw new globalThis.Error("Value is larger than Number.MAX_SAFE_INTEGER");
  }
  return long.toNumber();
}
```

So @8de2fdb0 if you wanted to work on a PR that did similar runtime detection of "incoming bigint will be larger than fixed64", that would sgtm.

I think it would be around in here:

https://github.com/stephenh/ts-proto/blob/main/src/main.ts#L1246

But not 100% sure. Thanks!

**Comment by 8de2fdb0** [comment]
I opened #938

**Comment by stephenh** [comment]
Fixed in #938 , thanks @8de2fdb0 !
